pkgs/top-level/all-packages.nix

```diff
@@ -2534,6 +2534,18 @@ in
     callPackage ../servers/search/elasticsearch/plugins.nix { }
   );
 
+  elasticsearch-curator = with (python3.override {
+    packageOverrides = self: super: {
+      click = super.click.overridePythonAttrs (oldAttrs: rec {
+        version = "6.7";
+        src = oldAttrs.src.override {
+          inherit version;
+          sha256 = "f15516df478d5a56180fbf80e68f206010e6d160fc39fa508b65e035fd75130b";
+        };
+      });
+    };
+  }).pkgs; toPythonApplication elasticsearch-curator;
```

Comment:
Minor nitpick. `toPythonApplication` could be put into `packageOverrides`. This would remove usage of `with`:
```
elasticsearch-curator = let 
  packageOverrides = self: super: {
    elasticsearch-curator = super.toPythonApplication super.elasticsearch-curator;
    click = super.click...
  };
  customPythonPackages = (python3.overrde { inherit packageOverrides; }).pkgs;
in
  customPythonPackages.elasticsearch-curator;
```
It looks kinda miracle, that `elasticsearch-curator` didn't select `self.elasticsearch-curator` from outer scope...